src/ucs/config/parser.c

```diff
@@ -920,10 +920,6 @@ void ucs_config_release_allow_list(void *ptr, const void *arg)
 {
     ucs_config_allow_list_t *allow_list = ptr;
 
-    if (allow_list->mode == UCS_CONFIG_ALLOW_LIST_ALLOW_ALL) {
```

Comment:
aren't we already releasing the array in line 880?

Reply:
Yes we do. The comment is on outdated diff, I removed it in `280867d`.

Reply:
yes, a saw you removed. but seems like we should keep this 'if', otherwise it's double-free

Reply:
Why double free? AFAIU, the `free()` itself happens only once, deep in `ucs_config_release_array`.

Reply:
1. call ucs_config_release_array after detecting that the value is "all", in ucs_config_sscanf_allow_list
2. call ucs_config_release_array from ucs_config_release_allow_list